src/emitter.ts

```diff
@@ -0,0 +1,1126 @@
+    "load": "Event",
+    "loadstart": "Event",
+    "progress": "ProgressEvent",
+    "readystatechange": "ProgressEvent",
+    "resize": "UIEvent",
+    "timeout": "ProgressEvent"
+};
+
+// Used to decide if a member should be emitted given its static property and
+// the intended scope level.
+function matchScope(scope: EmitScope, x: Browser.Method) {
+    return scope === EmitScope.All || (scope === EmitScope.StaticOnly) === !!x.static;
+}
+
+/// Parameter cannot be named "default" in JavaScript/Typescript so we need to rename it.
+function adjustParamName(name: string) {
+    return tsKeywords.has(name) ? `_${name}` : name;
+}
+
+function getElements<K extends string, T>(a: Record<K, Record<string, T>> | undefined, k: K): T[] {
+    return a ? mapToArray(a[k]) : [];
+}
+
+function createTextWriter(newLine: string) {
+    let output: string;
+    let indent: number;
+    let lineStart: boolean;
+    let stack: { content: string, indent: number }[] = [];
+
+    const indentStrings: string[] = ["", "    "];
+    function getIndentString(level: number) {
+        if (indentStrings[level] === undefined) {
+            indentStrings[level] = getIndentString(level - 1) + indentStrings[1];
+        }
+        return indentStrings[level];
+    }
+
+    function write(s: string) {
+        if (lineStart) {
+            output += getIndentString(indent);
+            lineStart = false;
+        }
+        output += s;
+    }
+
+    function reset(): void {
+        output = "";
+        indent = 0;
+        lineStart = true;
+        stack = [];
+    }
+
+    function writeLine() {
+        if (!lineStart) {
+            output += newLine;
+            lineStart = true;
+        }
+    }
+
+    reset();
+
+    return {
+
+        reset: reset,
+
+        resetIndent() { indent = 0; },
+        increaseIndent() { indent++; },
+        decreaseIndent() { indent--; },
+
+        print: write,
+        printLine(c: string) { writeLine(); write(c); },
+
+        printWithAddedIndent(c: string) { this.increaseIndent(); this.printLine(c); this.decreaseIndent(); },
+
+        clearStack() { stack = []; },
+        stackIsEmpty() { return stack.length === 0; },
+        printLineToStack(content: string) { stack.push({ content, indent }); },
+        printStackContent() {
+            stack.forEach(e => {
+                const oldIndent = indent;
+                indent = e.indent;
+                this.printLine(e.content);
+                indent = oldIndent;
+            });
+        },
+
+        getResult() { return output; }
+    };
+}
+
+function isEventHandler(p: Browser.Property) {
+    return p.type === "EventHandlerNonNull" || p.type === "EventHandler";
+}
+
+export function emitWebIDl(webidl: Browser.WebIdl, flavor: Flavor) {
+    // Global print target
+    const printer = createTextWriter("\n");
+
+    const pollutor = getElements(webidl.interfaces, "interface").find(i => flavor === Flavor.Web ? !!i["primary-global"] : !!i.global);
```

Comment:
Why is this called `pollutor`? Looks like the global interface?

Reply:
the spec refers to it as such. it is the object that "pollutes" the global namespace with its properties.